lib/gooddata/lcm/actions/synchronize_ldm.rb

```diff
@@ -35,57 +35,79 @@ class SynchronizeLdm < BaseAction
 
         description 'Allows to have facts with higher precision decimals'
         param :exclude_fact_rule, instance_of(Type::BooleanType), required: false, default: false
+
+        description 'Specifies how to synchronize LDM and resolve possible conflicts'
+        param :synchronize_ldm, instance_of(Type::SynchronizeLDM), required: false, default: 'diff_against_master_with_fallback'
       end
 
       class << self
         def call(params)
-          include_ca = params.include_computed_attributes.to_b
           exclude_fact_rule = params.exclude_fact_rule.to_b
-
+          client = params.gdc_gd_client
           results = []
+          synchronize = params.synchronize.map do |segment_info|
+            sync_segment(client, exclude_fact_rule, params, results, segment_info)
+          end
 
-          client = params.gdc_gd_client
-          development_client = params.development_client
+          {
+            results: results,
+            params: {
+              synchronize: synchronize
+            }
+          }
+        end
+
+        private
 
-          synchronize = params.synchronize.map do |info|
-            from_project = info.from
-            to_projects = info.to
+        def sync_segment(client, exclude_fact_rule, params, results, segment_info)
+          from_pid = segment_info[:from]
+          from = params.development_client.projects(from_pid) || fail("Invalid 'from' project specified - '#{from_pid}'")
 
-            from = development_client.projects(from_project) || fail("Invalid 'from' project specified - '#{from_project}'")
-            params.gdc_logger.info "Creating Blueprint, project: '#{from.title}', PID: #{from.pid}"
+          GoodData.logger.info "Creating Blueprint, project: '#{from.title}', PID: #{from_pid}"
+          blueprint = from.blueprint(include_ca: params.include_computed_attributes.to_b)
 
-            blueprint = from.blueprint(include_ca: include_ca)
-            info[:to] = to_projects.pmap do |entry|
-              pid = entry[:pid]
-              to_project = client.projects(pid) || fail("Invalid 'to' project specified - '#{pid}'")
+          maql_diff = nil
+          diff_against = segment_info[:diff_ldm_against]
+          if diff_against && params[:synchronize_ldm].start_with?('diff_against_master')
```

Comment:
can we use constants for this? or actually write the "diff_agains_master_with_callback" string here.. this implementation makes it difficult to grep for "diff_against_master_with_callback" usages (for no real gain)

Reply:
in case we decide against constants, we should still consider using symbols instead of strings.

Reply:
I agree it's difficult to grep. This would be better I think: `['diff_against_master_with_callback', 'diff_against_master'].include?(params[:synchronize_ldm].downcase)`

I don't see how constants would help in this case. Can you share an example?

IMHO using symbols would only mean additional `to_s` and no real gain.

Reply:
symbols offer a different semantical meaning in ruby, as they are a singleton and all occurences of them refers to the same object. This is usually leveraged to denote the "name" of something, as opposed to an arbitrary blob of text (string).

The gain would be mostly adhering to the underlying semantical difference in the language.

http://www.randomhacks.net/2007/01/20/13-ways-of-looking-at-a-ruby-symbol/

Reply:
constants would be a more old school C-like approach in this case. Since we are using Ruby, symbols, which offer the same advantages (easy search both by grepping and by lexical analysis), should be preffered.